simple_repository_server/routers/simple.py

```diff
@@ -59,19 +59,40 @@ def get_response_format(
 
 
 def build_router(
-    repository: SimpleRepository,
+    resource_repository: SimpleRepository,
+    *,
     http_client: httpx.AsyncClient,
+    prefix: str = "/simple/",
+    repo_factory: typing.Optional[typing.Callable[..., SimpleRepository]] = None,
 ) -> APIRouter:
+    """
+    Build a FastAPI router for the given repository and http_client.
+
+    Note that for the simple end-points, the repository is an injected
+    dependency, meaning that you can add your own dependencies into the repository
+    (see the test_repo_dependency_injection for an example of this).
```

Comment:
I think this is missing some detail that could be explained upfront. If I understand correctly, signature of the repo factory must correspond to the parameters defined in prefix.

Reply:
#16 relates